osu.Game/Overlays/Notifications/ProgressNotification.cs

```diff
@@ -0,0 +1,198 @@
+using osu.Game.Graphics;
+using OpenTK;
+using OpenTK.Graphics;
+
+namespace osu.Game.Overlays.Notifications
+{
+    public class ProgressNotification : Notification, IHasCompletionTarget
+    {
+        private string text;
+
+        private float progress;
+        public float Progress
+        {
+            get { return progress; }
+            set
+            {
+                Debug.Assert(state == ProgressNotificationState.Active);
+                progress = value;
+                progressBar.Progress = progress;
+            }
+        }
+
+        public ProgressNotificationState State
+        {
+            get { return state; }
+            set
+            {
+                state = value;
+                switch (state)
+                {
+                    case ProgressNotificationState.Queued:
+                        Light.Colour = colourQueued;
+                        Light.Pulsate = false;
+                        progressBar.Active = false;
+                        break;
+                    case ProgressNotificationState.Active:
+                        Light.Colour = colourActive;
+                        Light.Pulsate = true;
+                        progressBar.Active = true;
+                        break;
+                    case ProgressNotificationState.Cancelled:
+                        Light.Colour = colourCancelled;
+                        Light.Pulsate = false;
+                        progressBar.Active = false;
+                        break;
+                }
+            }
+        }
+
+        private ProgressNotificationState state;
+
+        public Action Completed;
+
+        public override bool DisplayOnTop => false;
+
+        private ProgressBar progressBar;
+        private Color4 colourQueued;
+        private Color4 colourActive;
+        private Color4 colourCancelled;
+
+        public ProgressNotification(string text)
+        {
+            this.text = text;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colours)
+        {
+            colourQueued = colours.YellowDark;
+            colourActive = colours.Blue;
+            colourCancelled = colours.Red;
+
+            IconContent.Add(new Box
+            {
+                RelativeSizeAxes = Axes.Both,
+            });
+
+            Content.Add(new SpriteText
+            {
+                TextSize = 16,
+                Colour = OsuColour.Gray(128),
+                AutoSizeAxes = Axes.Y,
+                RelativeSizeAxes = Axes.X,
+                Text = text
+            });
+
+            NotificationContent.Add(progressBar = new ProgressBar
+            {
+                Origin = Anchor.BottomLeft,
+                Anchor = Anchor.BottomLeft,
+                RelativeSizeAxes = Axes.X,
+            });
+
+            State = ProgressNotificationState.Queued;
+        }
+
+        public void Complete()
+        {
+            state = ProgressNotificationState.Completed;

```

Comment:
Maybe add an assertion that we're not already completed here. Not sure how likely this is to be called twice, but better make sure.